### DISCUSSION TITLE
Powershell errorhandeling

### DISCUSSION THREAD

**Discussion by okajimo** [ImageMagick/ImageMagick#5640] [Help]
I am having trouble with error handeling in powershell.

```
try{
    identify -ping '.\notsupported.txt'
} catch {
    echo "error message"
}
```

i wrote  a simple script where i am expecting the catch to kick in but dosent work.
is there a way to check if files are compatebel in form of a if else statment or try catch?

**Comment by dlemstra** [comment]
You will need to check the exit code instead. No exception will be thrown.

**Reply by okajimo** [reply]
Do you by any chance have a example?
i can see it is listed in the documentation but i have no idea how i can access this in a variable or return it as a statuse.

if i could do a simple if ($status -eq 0 ) that would solve my problem.

https://imagemagick.org/script/exception.php

**Reply by dlemstra** [reply]
Your question is not ImageMagick specific. You should search somewhere else how you can check the exit code of a program in powershell.
